crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -90,6 +90,12 @@ impl<T: TransactionOrdering> TxPool<T> {
         }
     }
 
+    /// Retrieves the nonce of the last transaction for a given sender after processing all
+    /// transactions.
+    pub fn nonce_after_all_transactions(&self, sender: SenderId) -> Option<u64> {
```

Comment:
let's split this into two parts:

```
fn get_highest_transaction_by_sender() -> Option<Tx>

fn get_highest_nonce_by_sender()
```

we could even add companions to the txpool trait for these